backend/r2_storage.py

```
@@ -0,0 +1,195 @@
+            file_extension = Path(filename).suffix
+            unique_filename = f"{uuid.uuid4()}{file_extension}"
+            
+            # R2内のキー（パス）を生成
+            object_key = f"{folder}/{unique_filename}"
+            
+            # アップロード時のメタデータ
+            extra_args = {}
+            if content_type:
+                extra_args['ContentType'] = content_type
+            
+            # ACLをpublic-readに設定（公開アクセス可能にする）
```

Ruff (0.14.7)

Comment contains ambiguous `（` (FULLWIDTH LEFT PARENTHESIS). Did you mean `(` (LEFT PARENTHESIS)?

(RUF003)